indexd/drs/blueprint.py

```diff
@@ -0,0 +1,153 @@
+    from local_settings import settings
+except ImportError:
+    from indexd.default_settings import settings
+
+import requests
+
+blueprint = flask.Blueprint("drs", __name__)
+
+blueprint.config = dict()
+blueprint.index_driver = None
+blueprint.dist = []
+
+@blueprint.route("/ga4gh/drs/v1/objects/<path:object_id>", methods=["GET"])
+def get_drs_object(object_id):
+    """
+    Returns a specific DRSobject with object_id
+    """
+    ret = blueprint.index_driver.get(object_id)
+    return flask.jsonify(indexd_to_drs(ret)), 200
+
+    
+@blueprint.route("/ga4gh/drs/v1/objects", methods=["GET"])
+def list_drs_records():
+    start = flask.request.args.get("page_token")
+    limit = flask.request.args.get("page_size")
+
+    try:
+        limit = 100 if limit is None else int(limit)
+    except ValueError:
+        raise UserError("limit must be an integer")
+
+    if limit <= 0 or limit > 1024:
+        raise UserError("limit must be between 1 and 1024")
+
+    url = flask.request.args.get("url")
+
+    checksum = flask.request.args.get("checksum")
+    if checksum:
+        hashes = {checksum["type"]: checksum["checksum"]}
+    else:
+        hashes = None
+
+    records = blueprint.index_driver.ids(
+        start=start, limit=limit, urls=url, hashes=hashes
+    )
+
+    ret = {"drs_objects": [indexd_to_drs(record, True)["drs_object"] for record in records]}
+
+    return flask.jsonify(ret), 200
+
+
+@blueprint.route("/ga4gh/drs/v1/objects/<path:object_id>/access/<path:access_id>", methods=["GET"])
+def get_signed_url(object_id, access_id=""):
+    res = get_signed_url_for_object(object_id, access_id)
+    if not res:
+        raise IndexNoRecordFound("No signed url found")
+    return res, 200
+
+
+def indexd_to_drs(record, list_drs = False):
+    bearer_token = flask.request.headers.get('AUTHORIZATION')
+    drs_object = {"id": record["did"], "description": "", "mime_type": ""}
+    
+    if "file_name" in record:
+        drs_object["name"] = record["file_name"]
+
+    if "self_uri" in record:
+        drs_object["self_uri"] = record["self_uri"]
+
+    if "created_date" in record:
+        drs_object["created_time"] = record["created_date"]
+
+    if "updated_date" in record:
+        drs_object["updated_time"] = record["updated_date"]
+
+    if "size" in record:
+        drs_object["size"] = record["size"]
+
+    if "alias" in record:
+        drs_object["aliases"].append(record["alias"])
+
+    if "contents" in record:
+        drs_object["contents"] = record["contents"]
+
+    # access_methods mapping
+    if "urls" in record:
+        drs_object["access_methods"] = []
+        for location in record["urls"]:
+            location_type = location.split(":")[0] #(s3, gs, ftp, gsiftp, globus, htsget, https, file)
+
+            drs_object["access_methods"].append({
+                "type":location_type, 
+                "access_url":get_signed_url_for_object(record["did"], "") if bearer_token and not list_drs else {"url":location},
```

Comment:
do you not need `access_id` in this call to `get_signed_url_for_object`? if `access_id` is optional, i'd suggest using a default value in the function definition of `get_signed_url_for_object` instead of passing an empty string here

Reply:
Doing this because the endpoint `/ga4gh/drs/v1/objects/<path:object_id>` when given a bearer token should have presigned_url in the url field. Since we don't need an access_id for this endpoint but access_id is **required** for `/ga4gh/drs/v1/objects/<path:object_id>/access/<path:access_id>`